really-core/src/main/scala/io/really/gorilla/Replayer.scala

```diff
@@ -0,0 +1,205 @@
+import play.api.libs.json.JsObject
+import io.really.CommandError.ObjectNotFound
+import io.really.protocol.SubscriptionFailure
+
+import scala.concurrent.Future
+import scala.slick.lifted.TableQuery
+import scala.slick.driver.H2Driver.simple._
+import scala.util.control.NonFatal
+import scala.concurrent.ExecutionContext.Implicits.global
+import EventLogs._
+
+class Replayer(globals: ReallyGlobals, objectSubscriber: ActorRef, rSubscription: RSubscription,
+    maxMarker: Option[Revision])(implicit session: Session) extends Actor with ActorLogging with Stash {
+
+  val replayerId = s"Replayer ${rSubscription.pushChannel.path}$$${rSubscription.r}"
+
+  val min = rSubscription.rev
+  private[gorilla] val r = rSubscription.r
+
+  context.watch(objectSubscriber)
+  //TODO Watch Gorilla to notify the ObjectSubscriber that the subscription failed in case of Gorilla's termination
+
+  maxMarker.map {
+    max: Revision =>
+      if (min == max) {
+        context.become(servePushUpdates)
+      } else if (min > max && (min - max) <= globals.config.GorillaConfig.advancedRevisionLimit) {
+        context.become(waitingCorrectRevision)
+      } else if (min > max) {
+        shutdownWithObjectSubscriptionAck("Subscription revision advances the current known revision with more than" +
+          " the advancing limit")
+      } else {
+        getEvents(r, min)
+      }
+  }.getOrElse {
+    getEvents(r, min)
+  }
+
+  def commonHandler: Receive = {
+    case SubscriptionManager.Unsubscribe =>
+      shutdown("Received unsubscribe request")
+    case Terminated(`objectSubscriber`) =>
+      shutdown("Object Subscription Actor was stopped")
+    case msg =>
+      unexpectedMessage(msg.toString)
+  }
+
+  def receive: Receive = starterReceiver orElse commonHandler
+
+  def starterReceiver: Receive = {
+    case _: GorillaLogEntry =>
+      stash()
+    case SnapshotObject(obj) =>
+      //TODO State should have a revision field
+      val stateRev = (obj \ "_rev").as[Long]
+      if (min > stateRev && (min - stateRev) <= globals.config.GorillaConfig.advancedRevisionLimit) {
+        context.become(waitingCorrectRevision)
+      } else if (min > stateRev) {
+        shutdownWithObjectSubscriptionAck("Subscription revision advances the current known revision with more than" +
+          " the advancing limit")
+      } else {
+        objectSubscriber ! SnapshotObject(obj)
+        context.become(servePushUpdates)
+      }
+    case SnapshotFetchError(r, _) =>
+      val msg = "Failed to retrieve a snapshot for that object"
+      shutdownWithObjectSubscriptionAck(msg)
+  }
+
+  def shutdown(reason: String) = {
+    log.warning(s"$replayerId actor is going to die because of: $reason")
+    globals.mediator ! Unsubscribe(r.toString, self)
+    context.stop(self)
+  }
+
+  def shutdownWithObjectSubscriptionAck(reason: String) = {
+    shutdown(reason)
+    objectSubscriber ! SubscriptionFailure(r, 500, reason)
```

Comment:
Because the `ObjectSubscriber` should handle this with sending the failure to the `Websocket` first, I think I should switch these 2 lines order to send the message firs
t, OK?